chain/stmgr/call.go

```diff
@@ -159,6 +159,8 @@ func (sm *StateManager) callInternal(ctx context.Context, msg *types.Message, pr
 	if err != nil {
 		return nil, xerrors.Errorf("failed to set up vm: %w", err)
 	}
+	defer func() { vmi.Done() }()
```

Comment:
I'm probably just tired today, but why do we need this IIFE instead of just a `defer vmi.Done()`. I could see this if `vmi` was being replaced, but that doesn't seem to be the case?

Reply:
Oh. It _is_ the case. Could you add a comment here?

Reply:
done.

Reply:
maybe, but this clearly works in all paths and you dont have to double think.

In general i prefer to keep resource acquisition and deferred release adjacent, it makes for cleaner code.